lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,21 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
```

Comment:
I find important to notice here without having data into the queue we're might actually not testing the methods into the running, for example in the unix input:

```
   @client_threads << Thread.start(@server_socket.accept) do |s|
          puts("Accepted connection", :server => "#{@path}")
          handle_socket(s, output_queue)
        end
``` 
this code will never get into the handle socket as the queue will always be empty. What do you think?